xbmc/addons/interfaces/GUI/DialogKeyboard.cpp

```diff
@@ -0,0 +1,216 @@
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "ADDON::Interface_GUIDialogKeyboard::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  std::string str = &strNewPassword;
+  bool bRet = CGUIKeyboardFactory::ShowAndVerifyNewPassword(str, strHeading, allowEmptyResult, autoCloseMs);
+  if (bRet)
+    strncpy(&strNewPassword, str.c_str(), maxStringSize);
+  maxStringSize = str.length();
+  return bRet;
+}
+
+bool Interface_GUIDialogKeyboard::ShowAndVerifyNewPassword(void* kodiBase, char &strNewPassword, unsigned int &maxStringSize, unsigned int autoCloseMs)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "ADDON::Interface_GUIDialogKeyboard::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  std::string str = &strNewPassword;
+  bool bRet = CGUIKeyboardFactory::ShowAndVerifyNewPassword(str, autoCloseMs);
+  if (bRet)
+    strncpy(&strNewPassword, str.c_str(), maxStringSize);
+  maxStringSize = str.length();
+  return bRet;
+}
+
+int Interface_GUIDialogKeyboard::ShowAndVerifyPassword(void* kodiBase, char &strPassword, unsigned int &maxStringSize, const char *strHeading, int iRetries, unsigned int autoCloseMs)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "ADDON::Interface_GUIDialogKeyboard::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  std::string str = &strPassword;
+  int iRet = CGUIKeyboardFactory::ShowAndVerifyPassword(str, strHeading, iRetries, autoCloseMs);
+  if (iRet)
+    strncpy(&strPassword, str.c_str(), maxStringSize);
+  maxStringSize = str.length();
+  return iRet;
+}
+
+bool Interface_GUIDialogKeyboard::ShowAndGetFilter(void* kodiBase, char &aTextString, unsigned int &maxStringSize, bool searching, unsigned int autoCloseMs)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "ADDON::Interface_GUIDialogKeyboard::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  std::string str = &aTextString;
+  bool bRet = CGUIKeyboardFactory::ShowAndGetFilter(str, searching, autoCloseMs);
+  if (bRet)
+    strncpy(&aTextString, str.c_str(), maxStringSize);
+  maxStringSize = str.length();
+  return bRet;
+}
+
+bool Interface_GUIDialogKeyboard::SendTextToActiveKeyboard(void* kodiBase, const char *aTextString, bool closeKeyboard)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "ADDON::Interface_GUIDialogKeyboard::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  return CGUIKeyboardFactory::SendTextToActiveKeyboard(aTextString, closeKeyboard);
+}
+
+bool Interface_GUIDialogKeyboard::isKeyboardActivated(void* kodiBase)
```

Comment:
comes, still working on the gui parts, there are few parts who bring me grey hairs, during improvement ;-)